lib/AST/NameLookup.cpp

```diff
@@ -918,32 +918,38 @@ void MemberLookupTable::updateLookupTable(NominalTypeDecl *nominal) {
 }
 
 void NominalTypeDecl::addedMember(Decl *member) {
+  auto *vd = dyn_cast<ValueDecl>(member);
+  if (!vd)
+    return;
+
   // If we have a lookup table, add the new member to it.
-  if (LookupTable.getPointer()) {
-    LookupTable.getPointer()->addMember(member);
+  auto *lookup = LookupTable.getPointer();
+  if (lookup && isLookupTablePopulated()) {
+    if (hasLazyMembers()) {
```

Comment:
Does it make sense to just skip this check? Even if the nominal doesn't have lazy members, the extensions might.

Reply:
I don’t think that can happen because prepareLookupTable() visits all extension members if the nominal doesn’t have lazy members. This should not be a problem because this indicates the extension was imported and the type was parsed (which I think is pretty rare) or someone loaded all members of the type (which is something we want to minimize).

Reply:
It can happen today with the Clang Importer's handling of structs and enums, but I assume you want to move away from that too.